awsService/src/main/java/com/kovan/app/service/S3Service.java

```diff
@@ -0,0 +1,226 @@
+            throw new FileException("File upload failed", e);
+        }
+        return uniqueId;
+    }
+
+    private String determineFolder(String fileName) {
+
+        String fileExtension = getFileExtension(fileName);
+        return switch (fileExtension.toLowerCase()) {
+            case "pdf" -> "pdfs";
+            case "doc", "docx" -> "docs";
+            case "jpg", "jpeg", "png" -> "images";
+            case "txt" -> "texts";
+            default -> "others";
+        };
+    }
+
+    private String getFileExtension(String fileName) {
+        if (isBlank(fileName) || !fileName.contains(".")) {
+            throw new FileException("Invalid file name: " + fileName);
+        }
+        return fileName.substring(fileName.lastIndexOf('.') + 1);
+    }
+
+    public byte[] downloadFile(String id) {
+
+        String fileName = service.findDocumentById(id).getFileName();
+
+        GetObjectRequest getObjectRequest = GetObjectRequest.builder()
+                .bucket(bucketName)
+                .key(fileName)
+                .build();
+
+        try (ResponseInputStream<GetObjectResponse> inputStream = s3Client.getObject(getObjectRequest);
+             ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+
+            byte[] buffer = new byte[1024];
+            int bytesRead;
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                outputStream.write(buffer, 0, bytesRead);
+            }
+
+            log.info("File with id {} downloaded successfully from bucket {}", id, bucketName);
+            return outputStream.toByteArray();
+        } catch (IOException e) {
+            log.error("Error downloading file with id {} from bucket {}", id, bucketName, e);
+            throw new FileException("File download failed", e);
+        }
+    }
+
+    public String deleteFile(String id) {
+        String fileName = service.findDocumentById(id).getFileName();
+
+        DeleteObjectRequest deleteObjectRequest = DeleteObjectRequest.builder()
+                .bucket(bucketName)
+                .key(fileName)
+                .build();
+
+        try {
+            s3Client.deleteObject(deleteObjectRequest);
+            service.deleteFile(id);
+
+            log.info("File {} deleted successfully from bucket {}", fileName, bucketName);
+            return fileName + " removed ...";
+        } catch (Exception e) {
+            log.error("Error deleting file {} from bucket {}", fileName, bucketName, e);
+            throw new FileException("File deletion failed", e);
+        }
+    }
+
+    public List<String> listFiles() {
+        List<String> fileNames = new ArrayList<>();
+        try {
+            ListObjectsV2Request listObjectsV2Request = ListObjectsV2Request.builder()
+                    .bucket(bucketName)
+                    .build();
+
+            ListObjectsV2Response listObjectsV2Response = s3Client.listObjectsV2(listObjectsV2Request);
+
+            while (true) {
```

Comment:
please refactor this code not clear and not needed this while and for loop needs to be updated to stream, You are working on JDK 21 so not use JDK 7 feature

Reply:
Refactored the code.